pkg/commands/option/artifact.go

```diff
@@ -0,0 +1,84 @@
+package option
+
+import (
+	"github.com/afdesk/trivy-cli/pkg/commands/flags"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/xerrors"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// ArtifactOption holds the options for an artifact scanning
+type ArtifactOption struct {
+	Input      string
+	Timeout    time.Duration
+	ClearCache bool
+
+	SkipDirs    []string
+	SkipFiles   []string
+	OfflineScan bool
+
+	// this field is populated in Init()
+	Target string
+}
+
+// NewArtifactOption is the factory method to return artifact option
+func NewArtifactOption(c *cobra.Command) (ArtifactOption, error) {
+	input := viper.GetString(flags.FlagInput)
+
+	target := ""
+	if len(c.Flags().Args()) > 0 {
+		target = c.Flags().Args()[0]
+	}
+
+	if target == "" && input == "" {
+		return ArtifactOption{}, xerrors.New(`trivy requires at least 1 argument or --input option`)
+	}
```

Comment:
I want to know why it is moved in NewArtifactOption.

Reply:
it was just copied from `trivy`: https://github.com/aquasecurity/trivy/blob/67d94774ba85b94358f93f2a5bf6b3cae629397d/pkg/commands/option/artifact.go#L45-L49

Reply:
No, it is in Init in Trivy

Reply:
oh, yep. i understood